### PR TITLE
Added release and release_file parameters to pkg_rpm

### PR DESCRIPTION
**Pull request by mmikitka**
1. Added the "rpm_nvra" output which follows the recommended package naming convention of *Name-Version-Release.Arch.rpm* (NVRA). See http://ftp.rpm.org/max-rpm/ch-rpm-file-format.html

2. Expose "release" and "release_file" parameters to support the NVRA naming convention

3. Remove the mandatory constraint on version parameters (version or version_file): I want to auto-generate my SPEC file outside of the pkg_rpm rule, and I don't want pkg_rpm altering the file.

### CODE REVIEW

**Comment by leonpowis** [comment]
Thanks

File: tools/build_defs/pkg/rpm.bzl [hunk 1]
```diff
@@ -96,14 +124,21 @@ an RPM package based on the spec_file and data attributes.
 
 Args:
```
Comment (katre) [inline tools/build_defs/pkg/rpm.bzl]: Can you add documentation about the nvra output?

File: tools/build_defs/pkg/rpm.bzl [hunk 2]
```diff
@@ -30,8 +30,15 @@ def _pkg_rpm_impl(ctx):
     files += [ctx.file.version_file]
   elif ctx.attr.version:
     args += ["--version=" + ctx.attr.version]
-  else:
-    fail("Neither version_file nor version attribute was specified")
+
+  # Release can be specified by a file or inlined
```
Comment (katre) [inline tools/build_defs/pkg/rpm.bzl]: Nit: Please end this comment with a period to follow the common style.

File: tools/build_defs/pkg/rpm.bzl [hunk 3]
```diff
@@ -87,26 +118,37 @@ pkg_rpm = rule(
...
-        "rpm": "%{name}-%{architecture}.rpm",
-    },
+    outputs = _pkg_rpm_outputs,
     executable = False)
 """Creates an RPM format package from the data files.
 
 This runs rpmbuild (and requires it to be installed beforehand) to generate
 an RPM package based on the spec_file and data attributes.
 
+Two outputs are guaranteed to be produced: (1) "%{name}.rpm", and
```
Comment (katre) [inline tools/build_defs/pkg/rpm.bzl]: "Two outputs are guaranteed to be produced: "%{name}.rpm", and "%{name}-%{architecture}.rpm". If the "version" and "release" arguments are non-empty, a third output will be produced, following the RPM-recommended N-V-R.A format (Name-Version-Release.Architecture.rpm), "

**Comment by katre** [comment]
Looks good, I'll merge this internally. Might not finish until Monday, however.

**Comment by laszlocsomor** [comment]
@katre , did you manage to merge this? i'm pinging all open PRs

**Comment by katre** [comment]
I wasn't able to merge it because of the problem from #4747. Once that's resolved I can merge this one.

**Pull request closed without merging**